lib/Client/Util.php

```diff
@@ -26,9 +26,27 @@ public function randInt32() {
         return $this->_rng->int32();
     }
 
-    public function nowMicros() {
+    public static function nowMicros() {
         // Note: microtime returns the current time *in seconds* but with
         // microsecond accuracy (not the current time in microseconds!).
         return floor(microtime(TRUE) * 1000.0 * 1000.0);
     }
+
+    public function pushWithMax(&$arr, $item, $max) {
+        if (!($max > 0)) {
+            $max = 1;
+        }
+
+        $arr[] =  $item;
+
+        // Simplistic random discard
+        $count = count($arr);
+        if ($count > $max) {
+            $i = $this->randIntRange(0, $max - 1);
```

Comment:
I think it might be possible to replace the RNG in this file with calls directly to `mt_rand` (http://php.net/manual/en/function.mt-rand.php) and then make all of these calls static. If we use this directly in tests though then we probably want to maintain the ability to seed the generator. All in all probably outside of the scope here.